core/vidl/vidl_ffmpeg_istream_v0.9.txx

```diff
@@ -474,7 +474,7 @@ vidl_ffmpeg_istream::current_frame()
 
     // If the pixel format is not recognized by vidl then convert the data into RGB_24
     vidl_pixel_format fmt = vidl_pixel_format_from_ffmpeg(enc->pix_fmt);
-    if (fmt == VIDL_PIXEL_FORMAT_UNKNOWN)
+    if (fmt == VIDL_PIXEL_FORMAT_UNKNOWN || fmt == VIDL_PIXEL_FORMAT_YUV_420P)
```

Comment:
Your reasoning is solid.  I will remove it from the branch. 

> On Jan 12, 2016, at 3:44 PM, Matt Leotta <notifications@github.com> wrote:
> 
> In core/vidl/vidl_ffmpeg_istream_v0.9.txx <https://github.com/vxl/vxl/pull/98#discussion_r49510392>:
> 
> > @@ -474,7 +474,7 @@ vidl_ffmpeg_istream::current_frame()
> >  
> >      // If the pixel format is not recognized by vidl then convert the data into RGB_24
> >      vidl_pixel_format fmt = vidl_pixel_format_from_ffmpeg(enc->pix_fmt);
> > -    if (fmt == VIDL_PIXEL_FORMAT_UNKNOWN)
> > +    if (fmt == VIDL_PIXEL_FORMAT_UNKNOWN || fmt == VIDL_PIXEL_FORMAT_YUV_420P)
> The purpose of this block was to catch FFMPEG pixel formats that vidl does not know about or cannot represent. This is a fallback option which converts the image into a default format that vidl does understand (RGB_24). Otherwise the memory is wrapped (shallow copy) in a vidl_frame with no conversion. The purpose of this approach is to postpone format conversion until absolutely needed. By adding YUV_420P to this list you are forcing vidl to convert it early. If, for example, someone wanted to write that frame back out into a new video as YUV_420P, no conversion would be needed at all, but by adding this change every frame would be converted to RGB_24 and then back again.
> 
> If you were having problems it's probably due to a bug elsewhere in converting from YUV_420P. This is not the appropriate fix.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/vxl/vxl/pull/98/files#r49510392>.
>